### PR TITLE
add loglevel settings

### PR DESCRIPTION
**Pull request by hirosassa**
fixes https://github.com/m3dev/gokart/issues/211

Thanks to @TaylerUva

### CODE REVIEW

File: gokart/build.py [hunk 1]
```diff
@@ -9,14 +9,18 @@
...
         self.logger = getLogger(__name__)
         self.default_level = self.logger.level
+        self.log_level_in_build = level
 
     def __enter__(self):
-        if not self.verbose:
+        if self.log_level_in_build is not None:
+            logging.disable(self.log_level_in_build)
+            self.logger.setLevel(self.log_level_in_build)
+        elif not self.verbose:
```
Comment (hirosassa) [inline gokart/build.py:23]: [help wanted]
In this implementation, `log_level_in_build` is prioritized than `verbose`.
Which do you prefer this or `verbose` is prioritized one?

File: gokart/build.py [hunk 2]
```diff
@@ -9,14 +9,18 @@
 from gokart.utils import check_config, read_environ
 
 
-class HideLogger:
-    def __init__(self, verbose: bool):
+class LoggerConfig:
+    def __init__(self, verbose: bool, level: Optional[int]):
```
Comment (vaaaaanquish) [inline gokart/build.py:13]: [NITS] Maybe not int?

File: gokart/build.py (hunk 1 above)
Comment (vaaaaanquish) [reply gokart/build.py:23]: How do you think about remove `verbose`, and setting the default for `loglevel`?

```python
class LoggerConfig:
    def __init__(self, level: Optional[int]=logging.CRITICAL):
    ....

    def __enter__(self):
        if self.level==logging.CRITICAL:
            logging.disable(sys.maxsize)
        else:
            logging.disable(self.level)
        self.logger.setLevel(self.level)
        return self
```

File: test/test_build.py [hunk 3]
```diff
@@ -30,5 +32,16 @@ def test_build(self):
...
 
+class LoggerConfigTest(unittest.TestCase):
+    def test_logger_config(self):
+        verbose = False
+        level = None
+        for verbose, level, expected in ((False, None, logging.CRITICAL), (True, None, logging.NOTSET), (False, logging.DEBUG, logging.DEBUG),
+                                         (True, logging.DEBUG, logging.DEBUG)):
+            with self.subTest(verbose=verbose, level=level, expected=expected):
+                with LoggerConfig(verbose, level) as lc:
+                    self.assertEqual(lc.logger.level, expected)
```
Comment (vaaaaanquish) [inline test/test_build.py:43]: Thanks add unit test. 🆒

File: gokart/build.py (hunk 1 above)
Comment (vaaaaanquish) [reply gokart/build.py:23]: Backward compatibility will be lost, though. 
If you don't want to do that, I agree with @hirosassa 's  implementation.

**Comment by vaaaaanquish** [comment]
Adding more interfaces to parts that are already configurable is likely to lead to bugs.  
We need to think about whether `gokart.build` should be in charge of this.
https://github.com/m3dev/gokart/issues/211#issuecomment-860014657

**Comment by TaylerUva** [comment]
Any updates on this?

File: gokart/build.py (hunk 1 above)
Comment (Hi-king) [reply gokart/build.py:23]: I'm standing with @vaaaaanquish 's suggestion. Maintaining both verbose and level options are confusing to me.

Comment (Hi-king) [reply gokart/build.py:23]: I'm standing with @vaaaaanquish 's suggestion. Maintaining both verbose and level options are confusing to me.

**Review by Hi-king — CHANGES_REQUESTED** [review]

File: gokart/build.py (hunk 1 above)
Comment (hirosassa) [reply gokart/build.py]: @vaaaaanquish @Hi-king Thanks for your fruitful FBs. fixed!

File: gokart/build.py [hunk 4]
```diff
@@ -9,16 +9,18 @@
...
         self.default_level = self.logger.level
+        self.level = level
 
     def __enter__(self):
-        if not self.verbose:
+        if self.level is not None:
+            logging.disable(self.level)
+            self.logger.setLevel(self.level)
+        if self.level == logging.CRITICAL:
             logging.disable(sys.maxsize)
```
Comment (TaylerUva) [inline gokart/build.py:23]: Why disable if set to critical?

Wouldn't the following be better?
```suggestion
        if self.level is None:
            logging.disable(sys.maxsize)
```

**Review by TaylerUva — CHANGES_REQUESTED** [review]

File: gokart/build.py (hunk 4 above)
Comment (hirosassa) [reply gokart/build.py]: Thanks! Your suggestion is right! Fixed!

**Comment by hirosassa** [comment]
@vaaaaanquish @Hi-king plz review!

File: gokart/build.py [hunk 5]
```diff
@@ -9,16 +9,19 @@
 from gokart.utils import check_config, read_environ
 
 
-class HideLogger:
-    def __init__(self, verbose: bool):
-        self.verbose = verbose
+class LoggerConfig:
+    def __init__(self, level: Optional[int]):
```
Comment (Hi-king) [inline gokart/build.py:13]: @hirosassa How about     `def __init__(self, level: int = logging.CRITICAL): ` instead of Optional ?

Comment (Hi-king) [reply gokart/build.py:13]: Are there any special reason to allow None instead of CRITICAL?

Comment (hirosassa) [reply gokart/build.py]: There's no specific reason to allow that. Fixed in  60f7a65.

**Review by vaaaaanquish — APPROVED** [review]
LGTM. thx

**Comment by vaaaaanquish** [comment]
@hirosassa plz write documents!

File: docs/intro_to_gokart.rst [hunk 6]
```diff
@@ -147,3 +147,13 @@ The :func:`~gokart.build` is inline code.
...
+
+
+To output logs of each tasks, you can pass `~log_level` parameter to `~gokart.build` as follows:
+
+.. code:: python
+
+    gokart.build(SampleTask(param='hello'), return_value=False, log_level=logginge.DEBUG)
+
+
+This feature is very useful for running `~gokart` on jupyter notebook.
```
Comment (hirosassa) [inline docs/intro_to_gokart.rst:159]: @vaaaaanquish docs added!

**Comment by hirosassa** [comment]
@Hi-king simplified in 12cd1d6. plz check this

**Review by Hi-king — APPROVED** [review]

**Comment by Hi-king** [comment]
@hirosassa Thx for continuous updates
@TaylerUva @vaaaaanquish Thx for reviews

Merged :)

**Pull request merged**